### PR TITLE
escape get_the_title

### PR DESCRIPTION
**Pull request by bmcculley**
use esc_html function on get_the_title to prevent potential
malicious issues.

#### Changes proposed in this Pull Request:

Updated comments.php added esc_html function on get_the_title to prevent potential
malicious issues.

#### Related issue(s):

issue #1366

### CODE REVIEW

**Comment by Ismail-elkorchi** [comment]
@bmcculley Thank you for your pull request.

In addition to `comments.php`, `inc/template-tags.php`, `template-parts/content-page.php` and `template-parts/content.php` also [contain instances](https://github.com/Automattic/_s/search?q=get_the_title&type=Code) of  the `get_the_title()` function which needs to be escaped.

Could you please update your pull request to escape those too?

**Comment by bmcculley** [comment]
@Ismail-elkorchi Thank you, I have updated the other files as well.

**Comment by bmcculley** [comment]
@tomjn should I take this as the final word and change it now or wait for more input?

**Comment by IanDelMar** [comment]
You could consider using the ['the_title' filter](https://developer.wordpress.org/reference/hooks/the_title/) instead of re-writing all the files. I know it's not late escaping but it makes it much more easier to change the then default escaping function sitewide.

**Comment by Ismail-elkorchi** [comment]
@bmcculley see https://github.com/Automattic/_s/issues/1366#issuecomment-605448941

File: template-parts/content-page.php [hunk 1]
```diff
@@ -41,7 +41,7 @@
 							),
 						)
 					),
-					get_the_title()
+					esc_html( get_the_title() )
```
Comment (carolinan) [inline template-parts/content-page.php]: wp_kses is already used here, on line 35.

**Comment by Ismail-elkorchi** [comment]
@bmcculley the recommendation @tomjn gave you is right on point. Replacing `esc_html()` with `wp_kses_post()` is more appropriate in this case, as the title can contain some basic tags.

After making these changes, can you please squash your commits ?  [Here is how to do it](https://github.com/wprig/wprig/wiki/How-to-squash-commits). The PR then should be good to go.

File: template-parts/content-page.php (hunk 1 above)
Comment (Ismail-elkorchi) [reply template-parts/content-page.php]: This instance of `get_the_title()` is outside `wp_kses()` used on line 35.  So I think it is still relevant to escape it.

**Comment by bmcculley** [comment]
@Ismail-elkorchi, thank you. I have made and committed the changes.

**Comment by Ismail-elkorchi** [comment]
@bmcculley unfortunately some unrelated changes got included in the commit because your fork has gone out of sync. You should rebase your branch atop of the latest version of master.

**Comment by bmcculley** [comment]
@Ismail-elkorchi, sorry about that, rebased with master.

**Pull request merged**